pkg/controller/handler.go

```diff
@@ -128,12 +128,51 @@ func (c *Controller) handleVirtualService(ingress *networkingv1.Ingress) error {
 		uvs.ObjectMeta.Annotations = nvs.ObjectMeta.Annotations
 		uvs.Spec = nvs.Spec
 
-		_, err = c.istioclientset.NetworkingV1beta1().VirtualServices(ingress.Namespace).Update(ctx, uvs, metav1.UpdateOptions{})
+		vs, err = c.istioclientset.NetworkingV1beta1().VirtualServices(ingress.Namespace).Update(ctx, uvs, metav1.UpdateOptions{})
 		if err != nil {
 			return err
 		}
 	}
 
+	// Check if we have an IP address on the first gateway associated
+	// with the  Virtual Service and sync it to the Ingress.
+	oid := strings.Split(vs.Spec.Gateways[0], "/")
+	if len(oid) == 1 {
+		oid[1] = oid[0]
+		oid[0] = vs.Namespace
+	}
```

Comment:
Hmmm... one last thing. This doesn't handle the case of a Gateway in the same namespace as the VS.

Reply:
It should! That's when the `len(oid) == 1` case is for :)

Reply:
I don't think you can you need to use the append()

Reply:
I get what you're doing now, sorry, stupid brain lol

Reply:
In the case where the gateway is in the same namespace, the array should be `[gateway]`

In the case of `namespace/gateway`, the array should be `[namespace, gateway]`